### PR TITLE
test: Fix UTC assuming test

### LINKED ISSUE #3720: Test assumes system timezone is UTC
[This test](https://github.com/getsentry/sentry-python/blob/02d09346e6d070e03b828807d72485b6f23b2c11/tests/test_utils.py#L70-L73), added in https://github.com/getsentry/sentry-python/issues/3546, assumes that the system time is UTC. The test fails otherwise because Python interprets the timestamp as if it is using the system's local timezone, but we assert that it is interpreted in UTC.

### PR DESCRIPTION
**Pull request by BYK**
Fixes #3720.

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -69,8 +69,8 @@ def _normalize_distribution_name(name):
         ),  # UTC time
         (
             "2021-01-01T00:00:00.000000",
-            datetime(2021, 1, 1, tzinfo=timezone.utc),
-        ),  # No TZ -- assume UTC
+            datetime(2021, 1, 1).astimezone(timezone.utc),
+        ),  # No TZ -- assume local but convert to UTC
```
Comment (asottile-sentry) [inline tests/test_utils.py:73]: given the comment I actually think the implementation is incorrect and should be something like:

```python
try:
   dt = datetime.fromisoformat(s)
except ...:
    ...
else:
   if dt.tzinfo is None:
       return dt.replace(tzinfo=UTC)
   else:
       return dt.astimezone(UTC)
```

Comment (BYK) [reply tests/test_utils.py:73]: I wrote that comment 😅 And after reading the ISO 8601 spec, I think I was mistaken: 

https://en.wikipedia.org/wiki/ISO_8601#Local_time_(unqualified)
>If no UTC relation information is given with a time representation, the time is assumed to be in local time.

Now there's the obvious warning this being unsafe in the next sentence but I'm split between being spec-compliant or "doing the right thing" which may confuse people if they rely on the spec.

What's your take @asottile-sentry?

Comment (asottile-sentry) [reply tests/test_utils.py:73]: I suspect it won't matter as most should run their servers in UTC anyway -- so I guess we can go with the current implementation

it seems to only be used to sort breadcrumbs in the event payload (is that even necessary? surely relay or the frontend should sort those right? do we even need this function and the caller?)

**Review by asottile-sentry — APPROVED** [review]
maybe we can just delete this code?

File: tests/test_utils.py (hunk 1 above)
Comment (BYK) [reply tests/test_utils.py:73]: >surely relay or the frontend should sort those right?

Agree with this but maybe we're trying to save resources? It also seems like the side effect of converting a string to a python date might be something we are after here. That said that python date will then be converted back into an ISO 8601 timestamp so 🤷🏻 

Will ask ingest folks.

**Pull request merged**